jakarta.ls/src/main/java/io/microshed/jakartals/commons/SnippetRegistry.java

```diff
@@ -182,15 +182,43 @@ private static Snippet createSnippet(
             item.setLabel(label);
             item.setDetail(snippet.getDescription());
 			String insertText = getInsertText(snippet, false, lineDelimeter);
+
             item.setKind(CompletionItemKind.Snippet);
             item.setDocumentation(Either.forRight(createDocumentation(snippet, canSupportMarkdown, lineDelimeter)));
-            item.setFilterText(label);
-			item.setTextEdit(new TextEdit(replaceRange, insertText));
+			item.setFilterText(label);
+			
+			TextEdit textEdit = new TextEdit(replaceRange, insertText);
+			formatTextEdit(textEdit); // Fixes indentation on the lines based on tabs
+			item.setTextEdit(textEdit);
 			item.setInsertTextFormat(InsertTextFormat.Snippet);
             return item;
 		}).collect(Collectors.toList());
 	}
 
+	/**
+	 * Applies character based formatting
+	 * @param edit
+	 * @return
```

Comment:
remove the `@return` from the javadoc comment since it is a void method, or have the method return the edited `TextEdit` item